src/mach/fat.rs

```diff
@@ -94,10 +94,26 @@ impl FatArch {
         &bytes[start..end]
     }
 
+    /// Returns the cpu type
+    pub fn cputype(&self) -> CpuType {
+        self.cputype
+    }
+
+    /// Returns the cpu subtype with the capabilities removed
+    pub fn cpusubtype(&self) -> CpuSubType {
+        self.cpusubtype & !CPU_SUBTYPE_MASK
+    }
+
+    /// Returns the capabilities of the CPU
+    pub fn cpu_caps(&self) -> u32 {
+        (self.cpusubtype & CPU_SUBTYPE_MASK) >> 24
+    }
+
     /// Whether this fat architecture header describes a 64-bit binary
     pub fn is_64(&self) -> bool {
-        self.cputype == cputype::CPU_TYPE_X86_64 || self.cputype == cputype::CPU_TYPE_ARM64
+        (self.cputype & CPU_ARCH_ABI64) != 0
```

Comment:
Should this be = 0? If not, I’d prefer the number it’s supposed to be equal to for being 64 bit

Reply:
I guess the check should be `(self.cputype & CPU_ARCH_ABI64) == CPU_ARCH_ABI64`?